workbench/src/menu.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2017 LarsGit223
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+/*
+ * Code for the "Workbench" menu.
+ */
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include "wb_globals.h"
+#include "dialogs.h"
+#include "menu.h"
+#include "sidebar.h"
+
+extern GeanyPlugin *geany_plugin;
+
+typedef struct
+{
+	GtkWidget *menu;
+	GtkWidget *root_item;
+	GtkWidget *item_new;
+	GtkWidget *item_open;
+	GtkWidget *item_save;
+	GtkWidget *item_settings;
+	GtkWidget *item_close;
+}WB_MENU_DATA;
+static WB_MENU_DATA menu_data;
+
+
+/* The function handles the menu item "New workbench" */
+static void item_new_workbench_activate_cb(GtkMenuItem *menuitem, gpointer user_data)
+{
+	gchar *filename;
+	GError *error = NULL;
+
+	filename = dialogs_create_new_workbench();
+	if (filename == NULL)
+	{
+		return;
+	}
+	wb_globals.opened_wb = workbench_new();
+	workbench_set_filename(wb_globals.opened_wb, filename);
+	if (workbench_save(wb_globals.opened_wb, &error))
+	{
+		menu_item_new_deactivate();
+		menu_item_open_deactivate();
+		menu_item_save_activate();
+		menu_item_settings_activate();
+		menu_item_close_activate();
```

Comment:
Yes. Just changed it to work in a similar way like the sidebar and the popup menu.